app/javascript/components/util/DateInput.js

```diff
@@ -3,34 +3,54 @@ import PropTypes from 'prop-types';
 import { OverlayTrigger, Tooltip } from 'react-bootstrap';
 import DatePicker from 'react-datepicker';
 import MaskedInput from 'react-text-mask';
-
 import moment from 'moment';
 
 class DateInput extends React.Component {
   constructor(props) {
     super(props);
     this.datePickerRef = React.createRef();
-    this.handleRawChange = this.handleRawChange.bind(this);
-    this.handleDateChange = this.handleDateChange.bind(this);
-    this.clearDate = this.clearDate.bind(this);
+    this.state = {
+      lastValidDate: props.date || null,
+    };
   }
 
-  handleDateChange(date) {
-    this.props.onChange(date && moment(date).format('YYYY-MM-DD'));
+  handleDateChange = date => {
+    const momentDate = date && moment(date).format('YYYY-MM-DD');
+    this.props.onChange(momentDate);
     this.datePickerRef.current.setOpen(false);
-  }
+    if (this.validDate(momentDate)) {
+      this.setState({ lastValidDate: momentDate });
+    }
+  };
 
-  handleRawChange(event) {
+  handleRawChange = event => {
     if (event.target.value) {
       this.datePickerRef.current.setOpen(true);
     }
+  };
+
+  handleOnBlur = () => {
+    // if date is not valid when clicking out of the date input
+    if (!this.validDate(this.props.date)) {
+      // change back to the last valid date or today if the field is required
+      if (this.props.required) {
```

Comment:
Could we change this prop? `required` to me doesn't indicate "returns to last valid date when invalid date is entered". Maybe like `replaceInvalid`?

Reply:
done here: https://github.com/SaraAlert/SaraAlert/pull/734/commits/5f20fac4fecdfd502e965f953c798387862c84e2

Reply:
thought about this some more and I think `required` is the right variable name because the goal for the advanced filter date inputs it to catch an empty input and put the last valid value in there. Open to other name suggestions as well.  Thoughts?

Reply:
maybe `replaceBlank`? not sure

Reply:
I updated it to this